modules/edu.gemini.seqexec.engine/src/test/scala/edu/gemini/seqexec/engine/packageSpec.scala

```diff
@@ -333,4 +315,86 @@ class packageSpec extends FlatSpec with NonImplicitAssertions {
     }
   }
 
+  it should "skip steps marked to be skipped at the beginning of the sequence." in {
+    val s0: Engine.State = Engine.State(Conditions.default,
+      None,
+      Map((seqId, Sequence.State.init(Sequence(
+        "First",
+        SequenceMetadata(GmosS, None, ""),
+        List(
+          Step.step(1, None, config, Set(GmosS), executions).copy(skipMark = true),
+          Step.step(2, None, config, Set(GmosS), executions),
+          Step.step(3, None, config, Set(GmosS), executions)
+        )
+      ) ) ) )
+    )
+
+    val sf = runToCompletion(s0)
+
+    inside (sf.map(_.sequences(seqId).done.map(Step.status))) {
+      case Some(stepSs) => assert(stepSs === List(StepState.Skipped, StepState.Completed, StepState.Completed))
+    }
+  }
+
+  it should "skip steps marked to be skipped in the middle of the sequence." in {
```

Comment:
What about a test with several skipped steps?

Reply:
Ok, I added one test for that.